File: octgnFX/agsxmpp/trunk/samples/VS2005/MiniClient/frmSubscribe.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;

using agsXMPP;
using agsXMPP.protocol.client;

namespace MiniClient
{
	/// <summary>
	/// Zusammenfassung für frmSubscribe.
	/// </summary>
	public class frmSubscribe : System.Windows.Forms.Form
	{
		private System.Windows.Forms.Button cmdApprove;
		private System.Windows.Forms.Button cmdRefuse;
		private System.Windows.Forms.Label label1;
		private System.Windows.Forms.Label lblFrom;
		/// <summary>
		/// Erforderliche Designervariable.
		/// </summary>
		private System.ComponentModel.Container components = null;

		private XmppClientConnection	_connection;
		private Jid						_from;

		public frmSubscribe(XmppClientConnection con, Jid jid)
		{
			InitializeComponent();
            
			_connection = con;
			_from		= jid;

			lblFrom.Text	= jid.ToString();
		}

		/// <summary>
		/// Die verwendeten Ressourcen bereinigen.
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if(components != null)
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Vom Windows Form-Designer generierter Code
		/// <summary>
		/// Erforderliche Methode für die Designerunterstützung. 
		/// Der Inhalt der Methode darf nicht mit dem Code-Editor geändert werden.
		/// </summary>
		private void InitializeComponent()
		{
			this.cmdApprove = new System.Windows.Forms.Button();
			this.cmdRefuse = new System.Windows.Forms.Button();
			this.label1 = new System.Windows.Forms.Label();
			this.lblFrom = new System.Windows.Forms.Label();
			this.SuspendLayout();
			// 
			// cmdApprove
			// 
			this.cmdApprove.FlatStyle = System.Windows.Forms.FlatStyle.System;
			this.cmdApprove.Location = new System.Drawing.Point(168, 56);
			this.cmdApprove.Name = "cmdApprove";
			this.cmdApprove.Size = new System.Drawing.Size(72, 24);
			this.cmdApprove.TabIndex = 0;
			this.cmdApprove.Text = "Approve";
			this.cmdApprove.Click += new System.EventHandler(this.cmdApprove_Click);
			// 
			// cmdRefuse
			// 
			this.cmdRefuse.FlatStyle = System.Windows.Forms.FlatStyle.System;
			this.cmdRefuse.Location = new System.Drawing.Point(56, 56);
			this.cmdRefuse.Name = "cmdRefuse";
			this.cmdRefuse.Size = new System.Drawing.Size(72, 24);
			this.cmdRefuse.TabIndex = 1;
			this.cmdRefuse.Text = "Refuse";
			this.cmdRefuse.Click += new System.EventHandler(this.cmdRefuse_Click);
			// 
			// label1
			// 
			this.label1.Font = new System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
			this.label1.Location = new System.Drawing.Point(8, 16);
			this.label1.Name = "label1";
			this.label1.Size = new System.Drawing.Size(48, 16);
			this.label1.TabIndex = 2;
			this.label1.Text = "From:";
			// 
			// lblFrom
			// 
			this.lblFrom.Location = new System.Drawing.Point(56, 16);
			this.lblFrom.Name = "lblFrom";
			this.lblFrom.Size = new System.Drawing.Size(224, 32);
			this.lblFrom.TabIndex = 3;
			this.lblFrom.Text = "jid";
			// 
			// frmSubscribe
			// 
			this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
			this.ClientSize = new System.Drawing.Size(292, 88);
			this.Controls.Add(this.lblFrom);
			this.Controls.Add(this.label1);
			this.Controls.Add(this.cmdRefuse);
			this.Controls.Add(this.cmdApprove);
			this.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedSingle;
			this.MaximizeBox = false;
			this.MinimizeBox = false;
			this.Name = "frmSubscribe";
			this.Text = "Subscription Request";
			this.ResumeLayout(false);

		}
		#endregion

		private void cmdRefuse_Click(object sender, System.EventArgs e)
		{			
			PresenceManager pm = new PresenceManager(_connection);
			pm.RefuseSubscriptionRequest(_from);
	
			this.Close();
		}

		private void cmdApprove_Click(object sender, System.EventArgs e)
		{
			PresenceManager pm = new PresenceManager(_connection);
			pm.ApproveSubscriptionRequest(_from);			

			this.Close();
		}
	}
}

File: octgnFX/agsxmpp/trunk/samples/VS2003/MiniClientVB/frmSubscribe.vb
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Forms

Imports agsXMPP
Imports agsXMPP.protocol.client

Namespace MiniClient

	Public Class frmSubscribe
	Inherits System.Windows.Forms.Form
		Private components As System.ComponentModel.Container = Nothing
		Private _connection As XmppClientConnection
		Private _from As Jid
		Private label1 As System.Windows.Forms.Label
		Private cmdApprove As System.Windows.Forms.Button
		Private lblFrom As System.Windows.Forms.Label
		Private cmdRefuse As System.Windows.Forms.Button
		
		Public Sub New(ByVal con As XmppClientConnection, ByVal jid As Jid)
			InitializeComponent
			_connection = con
			_from = jid
			lblFrom.Text = jid.ToString
		End Sub

		Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
			If disposing Then
				If Not (components Is Nothing) Then
					components.Dispose
				End If
			End If
			MyBase.Dispose(disposing)
		End Sub

		Private Sub InitializeComponent()
			Me.cmdRefuse = New System.Windows.Forms.Button
			Me.lblFrom = New System.Windows.Forms.Label
			Me.cmdApprove = New System.Windows.Forms.Button
			Me.label1 = New System.Windows.Forms.Label
			Me.SuspendLayout
			'
			'cmdRefuse
			'
			Me.cmdRefuse.FlatStyle = System.Windows.Forms.FlatStyle.System
			Me.cmdRefuse.Location = New System.Drawing.Point(56, 56)
			Me.cmdRefuse.Name = "cmdRefuse"
			Me.cmdRefuse.Size = New System.Drawing.Size(72, 24)
			Me.cmdRefuse.TabIndex = 1
			Me.cmdRefuse.Text = "Refuse"
			AddHandler Me.cmdRefuse.Click, AddressOf Me.cmdRefuse_Click
			'
			'lblFrom
			'
			Me.lblFrom.Location = New System.Drawing.Point(56, 16)
			Me.lblFrom.Name = "lblFrom"
			Me.lblFrom.Size = New System.Drawing.Size(224, 32)
			Me.lblFrom.TabIndex = 3
			Me.lblFrom.Text = "jid"
			'
			'cmdApprove
			'
			Me.cmdApprove.FlatStyle = System.Windows.Forms.FlatStyle.System
			Me.cmdApprove.Location = New System.Drawing.Point(168, 56)
			Me.cmdApprove.Name = "cmdApprove"
			Me.cmdApprove.Size = New System.Drawing.Size(72, 24)
			Me.cmdApprove.TabIndex = 0
			Me.cmdApprove.Text = "Approve"
			AddHandler Me.cmdApprove.Click, AddressOf Me.cmdApprove_Click
			'
			'label1
			'
			Me.label1.Location = New System.Drawing.Point(8, 16)
			Me.label1.Name = "label1"
			Me.label1.Size = New System.Drawing.Size(48, 16)
			Me.label1.TabIndex = 2
			Me.label1.Text = "From:"
			'
			'frmSubscribe
			'
			Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
			Me.ClientSize = New System.Drawing.Size(292, 88)
			Me.Controls.Add(Me.lblFrom)
			Me.Controls.Add(Me.label1)
			Me.Controls.Add(Me.cmdRefuse)
			Me.Controls.Add(Me.cmdApprove)
			Me.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedSingle
			Me.MaximizeBox = false
			Me.MinimizeBox = false
			Me.Name = "frmSubscribe"
			Me.Text = "Subscription Request"
			Me.ResumeLayout(false)
		End Sub

		Private Sub cmdRefuse_Click(ByVal sender As Object, ByVal e As System.EventArgs)
			Dim pm As PresenceManager = New PresenceManager(_connection)
			pm.RefuseSubscriptionRequest(_from)
			Me.Close
		End Sub

		Private Sub cmdApprove_Click(ByVal sender As Object, ByVal e As System.EventArgs)
			Dim pm As PresenceManager = New PresenceManager(_connection)
			pm.ApproveSubscriptionRequest(_from)
			Me.Close
		End Sub
	End Class 
End Namespace
